gnomad_qc/v4/annotations/generate_variant_qc_annotations.py

```diff
@@ -316,14 +336,152 @@ def run_generate_sib_stats(
+    :param sib_stats_ht: Table with sibling statistics.
+    :param impute_features: Whether to impute features using feature medians (this is
+        done by variant type).
+    :param n_partitions: Number of partitions to use for final annotated Table.
+    :return: Hail Table with all annotations needed for variant QC.
+    """
+    truth_data_ht = get_truth_ht()
+
+    ht = info_ht.transmute(**info_ht.info, **info_ht.allele_info)
+    trio_stats_ht = trio_stats_ht.select(
+        *[f"{a}_{group}" for a in ["n_transmitted", "ac_children"] for group in GROUPS]
+    )
+
+    logger.info("Annotating Table with trio and sibling stats and reference truth data")
+    ht = ht.annotate(
+        **trio_stats_ht[ht.key],
+        **sib_stats_ht[ht.key],
+        **truth_data_ht[ht.key],
+    )
+    tp_map = {
+        "transmitted_singleton": "n_transmitted",
+        "sibling_singleton": "n_sib_shared_variants",
+    }
+
+    # Filter to only variants found in high quality samples and are not lowqual.
+    ht = ht.filter((ht.AC_high_quality_raw > 0) & ~ht.AS_lowqual)
+    ht = ht.select(
+        "a_index",
+        "was_split",
+        *FEATURES,
+        **{tp: hl.or_else(ht[tp], False) for tp in TRUTH_DATA},
+        **{
+            f"{tp}_{group}": hl.or_else(
+                (ht[f"{n}_{group}"] == 1) & (ht.AC_high_quality_raw == 2),
+                False,
+            )
+            for tp, n in tp_map.items()
+            for group in GROUPS
+        },
+        fail_hard_filters=(ht.QD < 2) | (ht.FS > 60) | (ht.MQ < 30),
+        singleton=ht.AC_release_raw == 1,
+        ac_raw=ht.AC_high_quality_raw,
+        ac=ht.AC_release,
+        ac_unrelated_raw=ht.AC_unrelated_raw,
+    )
+
+    if impute_features:
+        ht = median_impute_features(ht, {"variant_type": ht.variant_type})
+
+    ht = ht.repartition(n_partitions, shuffle=False)
+    ht = ht.checkpoint(
+        hl.utils.new_temp_file("variant_qc_annotations", "ht"), overwrite=True
+    )
+
+    summary = ht.group_by(
+        *TRUTH_DATA, *[f"{tp}_{group}" for tp in tp_map for group in GROUPS]
+    ).aggregate(n=hl.agg.count())
+    logger.info("Summary of truth data annotations:")
+    summary.show(-1)
+
+    return ht
+
+
+def get_tp_ht_for_vcf_export(
+    ht: hl.Table,
+    transmitted_singletons: bool = False,
+    sibling_singletons: bool = False,
+) -> Dict[str, hl.Table]:
+    """
+    Get Tables with raw and adj true positive variants to export as a VCF for use in VQSR.
+
+    :param ht: Input Table with transmitted singleton and sibling singleton information.
+    :param transmitted_singletons: Whether to include transmitted singletons in the
+        true positive variants.
+    :param sibling_singletons: Whether to include sibling singletons in the true
+        positive variants.
+    :return: Dictionary of 'raw' and 'adj' true positive variant Tables.
+    """
+    if not transmitted_singletons and not sibling_singletons:
```

Comment:
Just adding to clarify what we are making. Want to make two files (there is also an option for a 3rd, but we probably won't test it, so no reason to make it).
- VCF of just transmitted singletons: These are determined by looking at the trios and identifying variants that are found in one parent and the child, and those two individuals are the only two in the callset with the variant.
- VCF of transmitted singletons (defined above) and sibling singletons: sibling singletons are determined by looking at two individuals that are siblings and identifying variants that are found in both siblings, and these two individuals are the only individuals in the callset with that variant.
- We can also export just sibling singletons, but it's not really worth it.

We use these as true positives in the variant QC training so that we can see what the features we use in the model look like on rare variants that we have high confidence in being real.

This if statement is just throwing an error because if you don't pick one of these you are just asking for an empty VCF